Assets/Scripts/StateMachines/BaseStateMachine.cs

```diff
@@ -0,0 +1,67 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public abstract class BaseStateMachine : MonoBehaviour
+{
+    // Current state of the state machine
+    internal BaseState currentState;
+
+    // Set currentState to the initial state of the implementing state machine (default is null)
+    /*
+    private protected void Start()
+    {
+        InitializeState();
+    }
+    */
+
+    // Call update logic function of state
+    private void Update()
+    {
+        if (currentState != null) {
+            currentState.UpdateFrame();
+        }
+    }
+
+    // Call update physics function of state
+    private void FixedUpdate() {
```

Comment:
I don't think we'll need a fixed update, since we're not dealing with any physics elements. Doesn't hurt to keep it for now though.